agent/proxycfg/manager.go

```diff
@@ -10,27 +11,26 @@ import (
 	"github.com/hashicorp/consul/agent/structs"
 )
 
+// ErrStopped signals that the manager instance has already been stopped and
+// can't be run again.
+var ErrStopped = errors.New("manager stopped")
+
+// CancelFunc is a type for a returned function that can be called to cancel a
+// watch.
+type CancelFunc func()
+
 // Manager is a component that integrates into the agent and manages Connect
-// proxy configuration state. This is used both for "managed" and "unmanaged"
-// proxies where "managed" refers to whether the _process_ itself is supervised.
+// proxy configuration state. This should not be confused with the deprecated
+// "managed proxy" concept where the agent supervises the actual proxy process.
+// proxycfg.Manager is oblivious to the distinction and manages state for any
+// service registered with Kind == connect-proxy.
 //
 // The Manager ensures that any Connect proxy registered on the agent has all
 // the state it needs cached locally via the agent cache. State includes
 // certificates, intentions, and service discovery results for any declared
-// upstreams.
+// upstreams. See package docs for more detail.
 type Manager struct {
-	// cache is the agent's cache instance that can be used to retreive, store and
-	// monitor state for the proxies.
-	cache *cache.Cache
-	// state is the agent's local state to be watched for new proxy registrations.
-	state *local.State
-	// source describes the current agent's identity, it's used directly for
-	// prepared query discovery but also indirectly as a way to pass current
-	// Datacenter name into other request types that need it. This is sufficient
-	// for now and cleaner than passing the entire RuntimeConfig.
-	source *structs.QuerySource
-	// logger is the agent's logger to be used for logging logs.
-	logger *log.Logger
+	cfg ManagerConfig
```

Comment:
Looks fine to me but just noting that another option is to embed it `type Manager struct { ManagerConfig }` so that you don't need to qualify with `.cfg.X` everywhere.